Library/Homebrew/system_command.rb

```diff
@@ -34,7 +34,7 @@ def self.run!(command, **options)
   end
 
   def run!
-    puts command.shelljoin.gsub(/\\=/, "=") if verbose? || ARGV.debug?
+    puts redacted_command.shelljoin.gsub('\=', "=").gsub('\*', "*") if verbose? || ARGV.debug?
```

Comment:
This `.gsub('\*', "*")` seems to general. What about instead making `redacted_command` act on the already `shelljoin`ed string so you don't need to worry about `*` being modified?

Reply:
There is another `shelljoin` inside `ErrorDuringExecution`.

How about `.gsub('\*\*\*\*\*\*', "******")`

Reply:
> There is another `shelljoin` inside `ErrorDuringExecution`.

That's fine, both should call the same function as they both want the same behaviour.

Reply:
Ok, I create a new function `redact_secrets` to redact the string after shelljoin.